commons-db/src/test/java/com/palantir/nexus/db/sql/SQLStringTest.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.nexus.db.sql;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class SQLStringTest {
+    @Test
+    public void testCanonicalizeString() {
+        List<String> testQuery = ImmutableList.of(
+                "insert foo into bar ; ",
+                "insert foo into bar;",
+                "insert  foo into bar;   ",
+                "/* UnregisteredSQLString */ insert foo into bar;",
+                "  /* UnregisteredSQLString */insert foo into bar",
+                "  /* UnregisteredSQLString */insert foo into bar ");
+        String canonicalQuery = "insert foo into bar";
+
+        testQuery.forEach(sql -> assertEquals(canonicalQuery, SQLString.canonicalizeString(sql)));
+    }
+
+    @Test
+    public void testCanonicalizeBatch() {
+        List<String> testBatch = ImmutableList.of(
+                "/* UnregisteredSQLString */ insert foo into bar; /* UnregisteredSQLString */insert foo into bar;",
+                "insert foo into bar; /* UnregisteredSQLString */ insert foo into bar");
+        String canconicalBatch = "insert foo into bar; insert foo into bar";
+
+        testBatch.forEach(sql -> assertEquals(canconicalBatch, SQLString.canonicalizeString(sql)));
+    }
+
+    @Test
+    public void testCanonicalizeStringNoSpaces() {
```

Comment:
nit: seeing a test with this name I'd guess it be an analogue of the one-query test cases with removeWhitespaceEntirely rather than the batch test.